v-next/hardhat-keystore/src/internal/types.ts

```diff
@@ -0,0 +1,34 @@
+export type Format = "hh-unencrypted-keystore";
+
+export interface UnencryptedKeystoreFile {
+  _format: Format;
+  version: number;
+  keys: {
+    [key: string]: string;
+  };
+}
+
+export interface Keystore {
+  listKeys(): Promise<string[]>;
+  hasKey(key: string): Promise<boolean>;
+  addNewValue(key: string, value: string): Promise<void>;
+  removeKey(key: string): Promise<void>;
+  readValue(key: string): Promise<string>;
+  toJSON(): UnencryptedKeystoreFile;
```

Comment:
The keystore itself takes responibility for converting into a KeystoreFile object where that object is serializable by `JSON.stringify`.